src/main/java/com/pubnub/api/endpoints/access/builder/GrantTokenBuilder.java

```diff
@@ -1,240 +1,75 @@
 package com.pubnub.api.endpoints.access.builder;
 
-import com.pubnub.api.PubNubException;
 import com.pubnub.api.UserId;
-import com.pubnub.api.callbacks.PNCallback;
 import com.pubnub.api.endpoints.access.GrantToken;
-import com.pubnub.api.endpoints.remoteaction.RemoteAction;
 import com.pubnub.api.models.consumer.access_manager.sum.SpacePermissions;
 import com.pubnub.api.models.consumer.access_manager.sum.UserPermissions;
 import com.pubnub.api.models.consumer.access_manager.v3.ChannelGrant;
 import com.pubnub.api.models.consumer.access_manager.v3.ChannelGroupGrant;
-import com.pubnub.api.models.consumer.access_manager.v3.PNGrantTokenResult;
 import com.pubnub.api.models.consumer.access_manager.v3.UUIDGrant;
-import org.jetbrains.annotations.NotNull;
 
-import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 
-public abstract class GrantTokenBuilder {
-
-    private GrantTokenBuilder() {
+public class GrantTokenBuilder extends AbstractGrantTokenBuilder<GrantTokenBuilder> {
 
+    public GrantTokenBuilder(GrantToken grantToken) {
+        super(grantToken);
     }
 
-    public abstract static class Base<T> implements RemoteAction<PNGrantTokenResult> {
-        protected final GrantToken grantToken;
-
-        public Base(GrantToken grantToken) {
-            this.grantToken = grantToken;
-        }
-
-        @Override
-        public PNGrantTokenResult sync() throws PubNubException {
-            return grantToken.sync();
-        }
-
-        @Override
-        public void async(@NotNull PNCallback<PNGrantTokenResult> callback) {
-            grantToken.async(callback);
-        }
-
-        @Override
-        public void retry() {
-            grantToken.retry();
-        }
-
-        @Override
-        public void silentCancel() {
-            grantToken.silentCancel();
-        }
-
-        public abstract T queryParam(Map<String, String> queryParam);
+    public GrantTokenBuilder ttl(Integer ttl) {
```

Comment:
What do you think about adding deprecation info here because we are enforcing ttl in grantToken(Integer ttl) ?